File: Frontend/studentAllFrontend/src/app/services/spind.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Spind} from "../entities/spind.model";
import {Observable} from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class SpindService {

  private baseUrl = 'http://localhost:8080/api'

  constructor(private httpClient: HttpClient) { }

  getAllSpinde(): Observable<Spind[]> {
    return this.httpClient.get<Spind[]>(`${this.baseUrl}/spind`);
  }

  getSpindById(id: number): Observable<Spind> {
    return this.httpClient.get<Spind>(`${this.baseUrl}/spind/${id}`)
  }

  createSpind(spind: Spind): Observable<Spind> {
    return this.httpClient.post<Spind>(`${this.baseUrl}/spind`, spind)
  }

  updateSpind(spind: Spind): Observable<Spind> {
    return this.httpClient.put<Spind>(`${this.baseUrl}/spind`, spind)
  }

  deleteSpind(id?: number): Observable<any> {
    return this.httpClient.delete<any>(`${this.baseUrl}/spind/${id}`)
  }
}

File: Backend/studentSpindBackend/src/main/java/example1/services/SpindService.java
package example1.services;

import example1.entities.Spind;

import java.util.List;
import java.util.Optional;

public interface SpindService {

    List<Spind> getAllSpinde();

    Optional<Spind> getSpindById(Long id);

    Spind createNewSpind(Spind spind);

    void deleteSpindById(Long id);

    Spind updateSpind(Spind spind);
}
